DOCS-6028 cross-org sharing (#86)
mulesoft/docs-exchange: modules/ROOT/pages/to-share-an-asset.adoc

click Access Management > Users to add them.

. Click an asset to show its details and click Share.
. In the Share menu, type a few letters of the person's name you want to share this asset with.
Or click a filter to list only users who have Viewer, Contributor, or Admin rights.
. Click Add.
. Repeat steps 2 and 3 to add additional users to share the asset with.

You can also use the Share menu to change who you are sharing the asset with or to stop sharing with one or more users.

== Share With an External Organization

Exchange administrators and asset administrators can grant viewer access for an asset to everyone in a chosen external organization.

The organization granting viewer access is the granting organization, and the organization that receives access is the receiving organization.

. The receiving organization adds the granting organization to its "External Access" list in Anypoint Access Management, as described in xref:access-management::external-org-access.adoc[External Organization Access].
+
Assets can only be shared with another organization if that organization has enabled receiving viewer access to assets from external organizations.
+

> . Login to your Exchange administrator account in the granting organization.

Reviewer: Can it be an asset administrator too, maybe is not needed to be specified (It's above).

Author: Good catch! I fixed this to say "Exchange administrator or asset administrator".

. Click an asset to show its details and click Share.
. In the Share menu, begin typing the name of the external organization you want to share the asset with.
. Select the organization you want to grant viewer access to.
. Repeat steps 4 and 5 to add additional organizations to share the asset with.

> When an asset is shared with an organization, email notification is sent to the maintainer of the organization's default "All Users" team. Usually this is the organization administrator.

Reviewer: The email is sent to `organization administrators`

Author: Thanks! Fixed.

> You can also use the Share menu to change who you are sharing the asset with or to stop sharing with one or more organizations.

Reviewer: Just asking: Should Share modal be more accurate? 🤔

Author: This is a good point; menus and modals are different. I'm not sure all users will know the word "modal", so I rewrote slightly to make it clearer.

= Share an Asset
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/
:page-aliases: to-share-an-asset-with-a-user.adoc

> Exchange administrators can share an asset with another user or with a team in a separate organization.

Reviewer 2: Exchange administrators can share an asset with another user within the same organization, or with the whole organization. They can also share an asset with an external organization.

>> We are not dealing with Teams in this release

Author: Thank you! Fixed.

= Share an Asset
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/
:page-aliases: to-share-an-asset-with-a-user.adoc

Exchange administrators and asset administrators can share an asset with another user in their organization, with the whole organization, or with an external organization.

Any Exchange user can click the *Shared with me* link in the left navigation menu of Exchange to show all assets shared with their user account and all assets shared with their organization.

== Share With a User

If the person is not in your organization, click *Access Management* > *Users* to add them.

. Click an asset to show its details.
. Click *Share*.

> . Begin typing the name of the person you want to share this asset with, and select that person in the list. You can optionally click a filter to show only users who have Viewer, Contributor, or Admin rights.

Reviewer: `You can optionally click a filter to show only users who have Viewer, Contributor, or Admin rights.`
I'm not sure if I understand correctly what you wrote but this is not possible, the only way of searching for an identity is by typing the name, we do not include any filter.

Author: You're exactly right. I assumed the existing text at https://docs.mulesoft.com/exchange/to-share-an-asset-with-a-user was correct, and it's not. It looks like this bad info has been here since before Sep 2018. The writer probably misunderstood the purpose of the Viewer/Contributor/Admin drop-down. Rewrote this text to fix it.

> . Click *Add*.

Reviewer: We should remove this, there is no add button, you can select the user by clicking in the autocomplete row, or pressing enter (it will add the first result to the list).

Author: Thank you, and I'm sorry about that! I walked through the UI to clarify this and removed the Add step.

. Repeat steps 3 and 4 to share with additional users.

> 

Reviewer: As a final step...
Should we include(in each way of sharing) clicking the `save` button? 
It sounds silly, but it is necessary to complete the process, (there are places on the platform that only by adding the operation is completed)

What do you think?

Author: That's not silly at all. As you say, we should document the "Save" step. I added it to the end of the instructions for each of the three sections.

You can also click *Share* to change who you are sharing the asset with or to stop sharing with one or more users.

== Share With Your Organization

> Exchange administrators and asset administrators can grant viewer access for an asset to everyone in their organization.

Reviewer: Not only viewer! You can add also admin and contributor permissions to everyone in your org 🙂

BTW: Should we include another step (in sharing with a user, and sharing with your organization), suggesting the user to change the default role(viewer)?
i.e.  3 - Select the desired role on the dropdown of the new user row.

Author: Good points! Updated to emphasize that viewer, contributor, and admin are options. As you suggest, I think we should have a step for selecting the viewer/contributor/admin, so I added that in the sections for sharing with a user and sharing with your org.

= Share an Asset
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/
:page-aliases: to-share-an-asset-with-a-user.adoc

> Exchange administrators can share an asset with another user in their organization, with the whole organization, or with an external organization.

Reviewer: Should we include Asset administrator? They also can shared their assets, without being an `Exchange Administrator`

Author: Good point. Updated this to include asset admins. Thank you!

== Share With a User

If the person is not in your organization,
click Access Management > Users to add them.

. Click an asset to show its details and click Share.
. In the Share menu, type a few letters of the person's name you want to share this asset with.
Or click a filter to list only users who have Viewer, Contributor, or Admin rights.
. Click Add.
. Repeat steps 2 and 3 to add additional users to share the asset with.

You can also use the Share menu to change who you are sharing the asset with or to stop sharing with one or more users.

> == Share With Your Organization

Reviewer: Recently news, That feature won't be in the release, please removed it

Author: OK, I removed "Share With Your Organization". Thank you for telling me.

+
Assets can only be shared with another organization if that organization has enabled receiving viewer access to assets from external organizations.
+
. Login to your Exchange administrator account in the granting organization.
. Click an asset to show its details and click Share.
. In the Share menu, begin typing the name of the external organization you want to share the asset with.
. Select the organization you want to grant viewer access to.
. Repeat steps 4 and 5 to add additional organizations to share the asset with.

When an asset is shared with an organization, email notification is sent to the maintainer of the organization's default "All Users" team. Usually this is the organization administrator.

You can also use the Share menu to change who you are sharing the asset with or to stop sharing with one or more organizations.

Any Exchange user can click the *Shared with me* link in the left navigation menu of Exchange to show all assets shared with their user account and all assets shared with their organization.

> Only internal users can request access to an API or gain the contributor or administrator roles. To grant these permissions to a user from another organization, invite the user to become an internal user of your organization, and then grant the contributor, administrator, or API access requestor permissions.

Reviewer: This is not possible: `To grant these permissions to a user from another organization, invite the user to become an internal user of your organization, and then grant the contributor, administrator, or API access requestor permissions`

It's ok to clarify the restrictions of request API access, nevertheless @kuntald: Should we clarify that external shared API's and Policies can't be imported from API manager?

Author: It's not possible to invite a user from another organization to join your own organization? Or it's not possible for a user in your own organization to get contributor, admin, or API access requestor permissions?

Reviewer: Sorry, I double-check that, you can invite a user from another organization to join your own organization. But at this point, the user will be moved from his organization to your organization. For the organization, "A", It is not worth to lose a user, just because the organization "B" wants to share him an asset.

I would not write it, I do not think it brings much value

Author: True, removing a user from the other org to join your org wouldn't be worth it. Can the user belong to both orgs?

Reviewer: As far as I know, no he cannot.

Author: Sara confirmed that is correct. Thank you for correcting that! I removed the part about granting these perms to a user from another org.

= Share an Asset
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/
:page-aliases: to-share-an-asset-with-a-user.adoc

Exchange administrators and asset administrators can share an asset with another user in their organization, with the whole organization, or with an external organization.

Any Exchange user can click the *Shared with me* link in the left navigation menu of Exchange to show all assets shared with their user account and all assets shared with their organization.

> == Share With a User

Reviewer 3: ```suggestion
== Share with a User
```

Reviewer 3: How would you feel about adding "Assets" to all of these "Share..." titles?
I know that topic nesting makes this kind of obvious, but...I don't know.... The editor in me wants MORE...lol....

Author: I agree, these are clearer with "Assets" in each heading. Fixed.

> If the person is not in your organization, click *Access Management* > *Users* to add them.

Reviewer 3: ```suggestion
If a user that you want to share an asset with is not in your organization, click *Access Management* > *Users* to add them:
```

. Click an asset to show its details.
. Click *Share*.
. Begin typing the name of the person you want to share this asset with, and select that person in the list.

> . Select whether to grant the user Viewer, Contributor, or Admin rights.

Reviewer 3: If these are the names of the rights and not just the types of rights, it is OK to use init. caps...especially when followed by "rights," in this case.
Note that this instance abbreviates to Admin, whereas the previous did not, which made me think they were more general than names.

> . Repeat steps 3 and 4 to share with additional users.

Reviewer 3: ```suggestion
. Repeat Steps 3 and 4 to share with additional users.
```
I always consider references to a specific numbered step a proper noun....

. Click *Save*.

You can also click *Share* to change who you are sharing the asset with or to stop sharing with one or more users.

> == Share With Your Organization

Reviewer 3: ```suggestion
== Share with Your Organization
```

> Exchange administrators and asset administrators can share an asset to everyone in their organization.

Reviewer 3: ```suggestion
Exchange administrators and asset administrators can share an asset with everyone in their organization.
```

. Click an asset to show its details.
. Click *Share*.
. Begin typing the name of your organization.

> . Select the list item for everyone at your organization.

Reviewer 3: This is just one step above; should it be here as well, or should both cases be two separate steps?
I'm guessing that our audiences do not need to be told that when they begin typing a name, a list of complete names based on the partial name appears?

Author: I decided to break it into two steps in all cases. I think you're right, it's better not to make the instructions longer by spelling out that the list of complete names appears.

> . Select whether to grant Viewer, Contributor, or Admin rights.

Reviewer 3: The previous step like this one adds "the user"; do you want to add "the whole organization" here?
```suggestion
. Select whether to grant all members of your organization Viewer, Contributor, or Admin rights.
```

. Click *Save*.

When an asset is shared with an organization, email notification is sent to the organization administrators.

> You can also click *Share* to stop sharing the asset with everyone at the organization.

Reviewer 3: ```suggestion
You can also click *Share* to stop sharing the asset with everyone at your organization.
```

Reviewer 3: Would you ever want to share with all but just a few people by clicking *Share* to omit those people, as in the other two tasks?

Author: Currently Exchange only supports adding a whole org or adding people one at a time, and doesn't support adding a whole org and then subtracting people one at a time.

> == Share With an External Organization

Reviewer 3: ```suggestion
== Share with an External Organization
```

> Exchange administrators and asset administrators can grant viewer access for an asset to everyone in a chosen external organization.

Reviewer 3: ```suggestion
Exchange administrators and asset administrators can grant viewer access to an asset to everyone in a chosen external organization.
```

> The organization granting viewer access is the granting organization, and the organization that receives access is the receiving organization.

Reviewer 3: Do we really need to clarify this?

Author: Rewrote to make this more brief.

> . The receiving organization adds the granting organization to its "External Access" list in Anypoint Access Management, as described in xref:access-management::external-organization-access.adoc[External Organization Access].

Reviewer 3: I don't think we need quotation marks here because the name is capitalized.

Author: Good point. Fixed.

+

> Assets can only be shared with another organization if that organization has enabled receiving viewer access to assets from external organizations.

Reviewer 3: I had to reread this, so I wonder why. 
Maybe it's the word "viewer"? We haven't specified that this is just viewing access yet, have we?

I wondered if we could say just "Assets can be shared with another organization only if that organization has enabled it."

In any case, move "only" to precede "if."

Author: Earlier in this section it was specified that only viewer access can be granted across orgs.

I think "enabled it" would be taken to mean adding the one specific org to the External Access list, and this paragraph refers to enabling receiving shares from any org at all, which is a separate option in the config menu, so I'm leaving this paragraph in.

I moved "only" to precede "if."

+

> . The granting organization adds the receiving organization to its "External Access" list in Anypoint Access Management, as described in xref:access-management::external-organization-access.adoc[External Organization Access].

Reviewer 3: DITTO prev. comment about quotation marks.
This statement follows the previous so closely, and cross-refers to the same location, that I wonder if we can say that each org. adds the other to their External Access lists?

Author: Much better! I merged these.

> . The granting organization adds the receiving organization to its "External Access" list in Anypoint Access Management, as described in xref:access-management::external-organization-access.adoc[External Organization Access].

Reviewer 4: For this piece, bi-directional external access makes it such that both orgs *must* have each other on their external access lists. This still seems to read like the recipient is whitelisting the sharing org, which is no longer the case.

IAM doesn't use "granting" or "receiving" org designations, so this would be an Exchange-only thing and is not present in the IAM piece of the UI. Food for thought!

Author: True, now that the external access lists are bi-directional, I think it makes more sense to merge those steps, so I rewrote to do so.

The granting/receiving distinction is one I invented just for these Exchange docs, and is not in the Exchange UI either. I don't think I could have explained cross-org sharing in Exchange clearly without it. The IAM docs probably don't need the distinction or the terms to explain cross-org sharing in IAM clearly, though.

> . Login to your Exchange administrator or asset administrator account in the granting organization.

Reviewer 3: GLOBALLY change "login" to "log in" (to log in...to).

Reviewer 3: How about "To grant access to an asset to an external organization, log in to your account as an administrator of the granting organization."
There was a lot of intro text but nothing to point directly to starting the granting task.

Author: Fixed "log in."

Good point. Added the "To grant access..." clarification.

> . Click an asset to show its details.

Reviewer 3: ```suggestion
. Click the asset to show its details.
```
We are sharing a specific asset, right?

. Click *Share*.
. Begin typing the name of the external organization you want to share this asset with.
. Select the organization in the list.

> . Repeat steps 6 and 7 to share with additional organizations.

Reviewer 3: DITTO prev edits to these same steps earlier.

Author: Fixed to make the instructions into two steps in all cases.

. Click *Save*.

When an asset is shared with an organization, email notification is sent to the organization administrators.

You can also click *Share* to change who you are sharing the asset with or to stop sharing with one or more organizations.

> Only internal users in the same organization as the asset can request access to an API or gain the contributor or administrator roles.

Reviewer 3: Instead of "Only internal users in the same organization as the asset" should we use shorter wording that specifies "the granting organization"?

Author: Yes, that is better. Fixed.

> === Using Assets Shared with External Organizations

Reviewer 3: This heading does not make clear who is using the assets: the grantor or the receiver.
Should it be something like "Using Assets Shared by a Granting Organization"?
My first reading was that this referred to a user in the granting org, rather than the receiving org.
Consider replacing "external" with "receiving" in the following text.

Reviewer 3: Should we have a sentence or two here that lists types of assets shared, or is that covered elsewhere?

Author: Replaced "external" with "receiving" and rewrote for clarity. A few weeks ago the Exchange team realized that the Exchange docs needed a list of types of assets that can be shared, so happily that is now on the front page of the Exchange docs.

> If a connector is shared with an external organization, then in Studio or Flow Designer a user in the external organization can search for the connector, configure it, and use it in a Mule application.

Reviewer 3: ```suggestion
If a connector is shared with an external organization, then a user in that organization can use Studio or Flow Designer to search for the connector, configure it, and use it in a Mule application.
```

> If an example or template is shared with an external organization, then in Studio or Flow Designer a user in the external organization can implement a Mule application based on the shared template.

Reviewer 3: ```suggestion
If an example or template is shared with an external organization, then a user in that organization can use Studio or Flow Designer to implement a Mule application based on the shared template.
```

> If a REST, SOAP, or HTTP API is created in Exchange and shared with an external organization, then API Manager cannot manage the API in the external organization. API Manager search results show only APIs from the external organization user's business group. In Studio a user in the external organization can use the API in the flow as a source for scaffolding an application.

Reviewer 3: Is this correct?
```suggestion
If a REST, SOAP, or HTTP API is created in Exchange in one organization and then shared with an external organization, API Manager cannot manage the API in that external organization. API Manager search results show only APIs from the external organization user's business group. However, a user in the external organization can use Studio to use the API in the flow, as a source for scaffolding an application.
```
The second sentence troubles me because it seems to unnecessarily complicate matters. We are describing sharing and use, right, and not so much searching and confirming? Also, does this sentence mean that API Manager shows only APIs already existing in the receiving organization, and not anything shared.

Reviewer 3: Looking forward, it appears that the focus of the topic has changed a bit, from just using the shared asset to using it but with restrictions in managing it with API Manager. Is that right?
This is pretty complicated!
I wonder if some sort of illustration might help.

Author: Quite complicated!

Rewrote to emphasize the use, and indicate that searching is secondary.

API Manager does show only APIs already existing in the receiving organization, and not anything shared.

Assets are shared with Exchange so that they can be used in other products. This final section talks about the various products that can and cannot use these shared assets, and in what ways.

> If a RAML fragment is created in Design Center and shared with an external organization, then in API Designer a user in the external organization can import the fragment and add it as a dependency in the new API specification. If the fragment is unshared, then the API specification in the external organization shows the same error that is shown if the fragment is deleted, "File not found."

Reviewer 3: DITTO previous sentence restructuring suggestions for the external user using <product> to <do this>.

Reviewer 3: Why are we documenting what happens if something is not shared, when that is the focus of this topic set?

Author: Rewritten for clarity.

By "unshared" I meant that the granting organization shares it and then later stops sharing it. Rewrote to clarify that.
